core/domain/classifier_services_test.py

```diff
@@ -253,3 +253,58 @@ def test_update_of_classifier_training_jobs(self):
             classifier_services.get_classifier_training_job_by_id(job_id))
         self.assertEqual(classifier_training_job.exp_id, exp_id)
         self.assertEqual(classifier_training_job.status, test_status)
+
+    def test_retrieval_of_classifier_from_exploration_attributes(self):
+        """Test the get_classifier_from_exploration_attributes method."""
+
+        with self.assertRaisesRegexp(Exception, (
+            'Entity for class ClassifierDataModel with id '
+            'fake_id not found')):
+            classifier_services.get_classifier_by_id('fake_id')
+
+        exp_id = u'1'
+        state_name = 'Home'
+        classifier_id = 'classifier_id1'
+        classifier_id = classifier_models.ClassifierDataModel.create(
+            classifier_id, exp_id, 1, state_name,
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'][
+                'algorithm_id'], [], 1)
+        classifier_models.ClassifierExplorationMappingModel.create(
+            exp_id, 1, state_name, classifier_id)
+        classifier = (
+            classifier_services.get_classifier_from_exploration_attributes(
+                exp_id, 1, state_name))
+        self.assertEqual(classifier.exp_id, exp_id)
+        self.assertEqual(classifier.state_name, state_name)
+        self.assertEqual(classifier.id, classifier_id)
+
+    def test_creation_of_classifier_exploration_mapping(self):
+        """Test the create_classifier_exploration_mapping method."""
+
+        exp_id = u'1'
+        state_name = 'Home'
```

Comment:
If you want to test this properly, I suggest using a state name that contains unicode chars.